docs: Remove references to to previous named ansible collection cloudalchemy

andygrunwald opened the pull request:
The repository is a migration from https://github.com/cloudalchemy/ansible-node-exporter. The previous repository has been marked as archived already.

gardar commented:
Thanks! Do you mind fixing the references to cloudalchemy in the other roles too?

andygrunwald commented:
@gardar Sure.
Most of the other references are demo code that includes things like

```yaml
- demo.cloudalchemy.org:9100
- influx.cloudalchemy.org:9100
```

Do you prefer to keep it like this or to change those entries to something like

```yaml
- demo.example.org:9100
- influx.example.org:9100
```

?

gardar commented:
Actually what I was thinking was just the role names in the examples, like you have already proposed for node_exporter but apparently it's just the node_exporter and blackbox_exporter that has it.
But removing all references to cloudalchemy (and circleci) is something that needs to be done sooner or later, so if you're up to it - go ahead :)

I do believe those demos now live at: https://demo.do.prometheus.io/

andygrunwald commented:
@gardar Done.

I have
* Removed references to https://github.com/cloudalchemy/demo-site/
* Fixed a few links to Prometheus docs
* Replaced cloudalchemy URLs with https://demo.do.prometheus.io/ (where appropriate - other URLs are replaced with example.org
* Removed a few README paragraphs related to upgrades that are not fitting anymore

Let me know what you think and if anything is missing.

SuperQ reviewed (approved):
Thanks!

gardar reviewed (approved):
Super! Thanks.
